Changes in module 'nf-core/minimap2/align'
--- modules/nf-core/minimap2/align/main.nf
+++ modules/nf-core/minimap2/align/main.nf
@@ -3,7 +3,7 @@
     label 'process_medium'
 
     // Note: the versions here need to match the versions used in the mulled container below and minimap2/index
-    conda "bioconda::minimap2=2.24 bioconda::samtools=1.14"
+    conda { params.conda_minimap2_env ?: "${projectDir}/envs/minimap2.yml" }
     container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
         'https://depot.galaxyproject.org/singularity/mulled-v2-66534bcbb7031a148b13e2ad42583020b9cd25c4:1679e915ddb9d6b4abda91880c4b48857d471bd8-0' :
         'quay.io/biocontainers/mulled-v2-66534bcbb7031a148b13e2ad42583020b9cd25c4:1679e915ddb9d6b4abda91880c4b48857d471bd8-0' }"

************************************************************
