Changes in module 'nf-core/minimap2/index'
--- modules/nf-core/minimap2/index/main.nf
+++ modules/nf-core/minimap2/index/main.nf
@@ -31,4 +31,16 @@
         minimap2: \$(minimap2 --version 2>&1)
     END_VERSIONS
     """
+
+    stub: 
+    """
+    touch ${fasta.baseName}.mmi
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        minimap2: \$(minimap2 --version 2>&1)
+    END_VERSIONS
+    """
+
+
 }

************************************************************
