Changes in module 'nf-core/minimap2/index'
--- modules/nf-core/minimap2/index/main.nf
+++ modules/nf-core/minimap2/index/main.nf
@@ -2,16 +2,16 @@
     label 'process_medium'
 
     // Note: the versions here need to match the versions used in minimap2/align
-    conda "bioconda::minimap2=2.24"
+    conda { params.conda_minimap2_env ?: "${projectDir}/envs/minimap2.yml" }
     container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
         'https://depot.galaxyproject.org/singularity/minimap2:2.24--h7132678_1' :
         'quay.io/biocontainers/minimap2:2.24--h7132678_1' }"
 
     input:
-    tuple val(meta), path(fasta)
+    path(fasta)
 
     output:
-    tuple val(meta), path("*.mmi"), emit: index
+    path("*.mmi")                 , emit: index
     path "versions.yml"           , emit: versions
 
     when:
@@ -31,4 +31,16 @@
         minimap2: \$(minimap2 --version 2>&1)
     END_VERSIONS
     """
+
+    stub: 
+    """
+    touch ${fasta.baseName}.mmi
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        minimap2: \$(minimap2 --version 2>&1)
+    END_VERSIONS
+    """
+
+
 }

************************************************************
